src/outbox/outboxReducer.js

```diff
@@ -17,7 +17,15 @@ const initialState = NULL_ARRAY;
 const messageSendStart = (state, action) => {
   const message = state.find(item => item.timestamp === action.outbox.timestamp);
   if (message) {
-    return state;
+    // If we find there is already a message in the outbox with the same
+    // timestamp, replace it with the new message from the action. This is used
+    // to update & send mesasges with `deferSending` as `true`.
+    return state.map(item => {
+      if (item.timestamp === action.outbox.timestamp) {
+        return action.outbox;
+      }
+      return item;
+    });
```

Comment:
This behavior is a bit peculiar, and it only really becomes possible to understand given the context of the next commit. (One way of thinking of that: it's presenting an API which that next commit is using, and the API is quite particular and starts really making sense only when the way the API is used becomes visible.) So for that reason it'd be best to just squash these commits together.